src/libraries/Microsoft.PowerFx.Json/FormulaValueJSON.cs

```diff
@@ -163,14 +163,12 @@ internal static FormulaValue TableFromJsonArray(JsonElement array, TableType tab
         {
             Contract.Assert(array.ValueKind == JsonValueKind.Array);
 
-            var records = new List<RecordValue>();            
-            bool isArray = tableType?._type.IsColumn == true;
-            FormulaType ft = isArray ? tableType.ToRecord().GetFieldType("Value") : tableType?.ToRecord();
```

Comment:
This check was incorrect, even if you get the type of single column table, `JsonElemnt.Kind` is still `Object` for each element and would conflict when calling `FromJson()` at line 171.

Reply:
Turns out, Check was needed but with some extra checking.